database/database.go

```diff
@@ -76,14 +77,29 @@ const _selectAVU = `
 	       created_on,
 	       modified_on
 	  FROM avus
+	  %s
+	UNION ALL
+	SELECT cast(avus.id as varchar),
+	       coalesce(avus.attribute, ''),
+	       coalesce(avus.value, ''),
+	       coalesce(avus.unit, ''),
+	       cast(aa.target_id as varchar),
+	       cast(aa.target_type as varchar),
+	       avus.created_by,
+	       avus.modified_by,
+	       avus.created_on,
+	       avus.modified_on
+	  FROM avus
+	  JOIN all_avus aa ON (avus.target_id = cast(aa.id as uuid) AND avus.target_type = 'avu')
```

Comment:
TIL that referring to a subquery column where a function is used in the select expression actually works despite the fact that Postgres reports the function name instead of the column name in its output.

Reply:
I think it's specifically `cast` that works for this, and other functions wouldn't. Using `cast()` the column names work as expected still, rather than showing `cast` as the header:

```
metadata=> with recursive all_avus as (select id, attribute, value, unit, target_id, target_type from avus where target_id = '9ebd2858-34cf-11e6-824a-000e1e0af2dc' and target_type = 'folder' UNION ALL select avus.id, avus.attribute, avus.value, avus.unit, aa.target_id, aa.target_type from avus join all_avus aa on (avus.target_id = aa.id and avus.target_type = 'avu')) select cast(id as varchar), id from all_avus;
                  id                  |                  id                  
--------------------------------------+--------------------------------------
 11b65ee6-6093-11e9-a7f5-5a03816fc427 | 11b65ee6-6093-11e9-a7f5-5a03816fc427
 11b536d8-6093-11e9-a7f5-5a03816fc427 | 11b536d8-6093-11e9-a7f5-5a03816fc427
 11b3f502-6093-11e9-a7f5-5a03816fc427 | 11b3f502-6093-11e9-a7f5-5a03816fc427
 11b6e294-6093-11e9-a7f5-5a03816fc427 | 11b6e294-6093-11e9-a7f5-5a03816fc427
 11b596e6-6093-11e9-a7f5-5a03816fc427 | 11b596e6-6093-11e9-a7f5-5a03816fc427
 11b46280-6093-11e9-a7f5-5a03816fc427 | 11b46280-6093-11e9-a7f5-5a03816fc427
 11b7379e-6093-11e9-a7f5-5a03816fc427 | 11b7379e-6093-11e9-a7f5-5a03816fc427
 11b5f690-6093-11e9-a7f5-5a03816fc427 | 11b5f690-6093-11e9-a7f5-5a03816fc427
 11b4ce28-6093-11e9-a7f5-5a03816fc427 | 11b4ce28-6093-11e9-a7f5-5a03816fc427
(9 rows)
```

Reply:
Ah, the subquery column reference is on the `id` column, which is a `cast` rather than a `coalesce`. Nevermind. 😊